bot/exts/halloween/candy_collection.py

```diff
@@ -1,39 +1,54 @@
-import functools
 import json
 import logging
-import os
 import random
-from typing import List, Union
+from pathlib import Path
+from typing import Union
 
 import discord
 from discord.ext import commands
 
 from bot.constants import Channels, Month
 from bot.utils.decorators import in_month
+from bot.utils.persist import make_persistent
 
 log = logging.getLogger(__name__)
 
-json_location = os.path.join("bot", "resources", "halloween", "candy_collection.json")
-
 # chance is 1 in x range, so 1 in 20 range would give 5% chance (for add candy)
 ADD_CANDY_REACTION_CHANCE = 20  # 5%
 ADD_CANDY_EXISTING_REACTION_CHANCE = 10  # 10%
 ADD_SKULL_REACTION_CHANCE = 50  # 2%
 ADD_SKULL_EXISTING_REACTION_CHANCE = 20  # 5%
 
+EMOJIS = dict(
+    CANDY="\N{CANDY}",
+    SKULL="\N{SKULL}",
+    MEDALS=(
+        '\N{FIRST PLACE MEDAL}',
+        '\N{SECOND PLACE MEDAL}',
+        '\N{THIRD PLACE MEDAL}',
+        '\N{SPORTS MEDAL}',
+        '\N{SPORTS MEDAL}',
+    ),
+)
+
 
 class CandyCollection(commands.Cog):
     """Candy collection game Cog."""
 
     def __init__(self, bot: commands.Bot):
         self.bot = bot
-        with open(json_location, encoding="utf8") as candy:
-            self.candy_json = json.load(candy)
-            self.msg_reacted = self.candy_json['msg_reacted']
-        self.get_candyinfo = dict()
-        for userinfo in self.candy_json['records']:
-            userid = userinfo['userid']
-            self.get_candyinfo[userid] = userinfo
+
+        self.json_file = make_persistent(Path("bot", "resources", "halloween", "candy_collection.json"))
+
+        with self.json_file.open() as fp:
+            candy_data = json.load(fp)
+
+        # The rank data
```

Comment:
```suggestion
```
The variable name `candy_records` seems self explanatory to my, which would make this comment unnecessary.